src/gurt/telemetry.c

```diff
@@ -1295,6 +1329,88 @@ d_tm_print_stats(FILE *stream, struct d_tm_stats_t *stats, int format)
 	fprintf(stream, ", samples: %lu]", stats->sample_size);
 }
 
+static int
+_reset_node(struct d_tm_context *ctx, struct d_tm_node_t *node)
+{
+	struct d_tm_metric_t	*metric_data = NULL;
+	struct d_tm_stats_t	*dtm_stats = NULL;
+	struct d_tm_histogram_t *dtm_histogram = NULL;
+	struct d_tm_shmem_hdr	*shmem = NULL;
+	int			 rc;
+
+	if (ctx == NULL || node == NULL)
+		return -DER_INVAL;
+
+	rc = validate_node_ptr(ctx, node, &shmem);
+	if (rc != 0)
+		return rc;
+
+	metric_data = conv_ptr(shmem, node->dtn_metric);
+	if (metric_data == NULL)
+		return -DER_METRIC_NOT_FOUND;
+
+	dtm_stats = conv_ptr(shmem, metric_data->dtm_stats);
+	dtm_histogram = conv_ptr(shmem, metric_data->dtm_histogram);
+	d_tm_node_lock(node);
+	memset(&metric_data->dtm_data, 0, sizeof(metric_data->dtm_data));
+	if (dtm_stats != NULL)
+		memset(dtm_stats, 0, sizeof(*dtm_stats));
+
+	if (dtm_histogram != NULL) {
+		int i;
+
+		for (i = 0; i < dtm_histogram->dth_num_buckets; i++) {
+			struct d_tm_node_t	*bucket;
+
+			bucket = dtm_histogram->dth_buckets[i].dtb_bucket;
+			_reset_node(ctx, bucket);
+		}
+	}
+
+	d_tm_node_unlock(node);
+	return DER_SUCCESS;
+}
+
+static void
+reset_node(struct d_tm_context *ctx, struct d_tm_node_t *node, int level,
+	   char *path, int format, int opt_fields, FILE *stream)
+{
+	char	*name = NULL;
+
+	if (node == NULL)
+		return;
+
+	name = d_tm_get_name(ctx, node);
+	if (name == NULL)
+		name = "(null)";
+
+	switch (node->dtn_type) {
+	case D_TM_LINK:
+		node = d_tm_follow_link(ctx, node);
+		reset_node(ctx, node, level, path, format, opt_fields, stream);
+		break;
+	case D_TM_DIRECTORY:
+	case D_TM_COUNTER:
+	case D_TM_TIMESTAMP:
+	case D_TM_TIMER_SNAPSHOT:
+	case (D_TM_TIMER_SNAPSHOT | D_TM_CLOCK_REALTIME):
+	case (D_TM_TIMER_SNAPSHOT | D_TM_CLOCK_PROCESS_CPUTIME):
+	case (D_TM_TIMER_SNAPSHOT | D_TM_CLOCK_THREAD_CPUTIME):
+	case D_TM_DURATION:
+	case (D_TM_DURATION | D_TM_CLOCK_REALTIME):
+	case (D_TM_DURATION | D_TM_CLOCK_PROCESS_CPUTIME):
+	case (D_TM_DURATION | D_TM_CLOCK_THREAD_CPUTIME):
+	case D_TM_GAUGE:
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment